overlord/servicestate/servicemgr.go

```diff
@@ -61,8 +61,16 @@ func Manager(st *state.State, runner *state.TaskRunner) *ServiceManager {
 
 	// TODO: undo handler
 	runner.AddHandler("quota-control", m.doQuotaControl, nil)
-
-	snapstate.AddAffectedSnapsByKind("quota-control", quotaControlAffectedSnaps)
+	AddAffectedQuotasByKind("quota-control", affectedQuotasForQuotaControl)
+	snapstate.AddAffectedSnapsByKind("quota-control", affectedSnapsForQuotaControl)
+
+	// We can't directly refer to the servicestate internals from snapstate,
+	// so this task encapsulate taking care of calling quotaUpdate
+	// with the correct setup. This task also supports proper handling of
```

Comment:
Sorry, I'm a bit slow here :-) It's still not clear to me what this is all about. Is it explaining that the doQuotaAddSnap cannot be implemented in snapstate (because it wouldn't be able to call into the servicestate methods it needs), and therefore it has been implemented in this package (servicestate)? Or is it about something else?

Reply:
Yea exactly, we cannot refer to the servicestate package from snapstate, because we depend on snapstate from servicestate which results in a cyclic dependency. I can try to clear up the explanation, or feel free to make any suggestions to the current explanation

Reply:
Maybe "The doQuotaAddSnap() handle would fit better in snapstate, but we cannot have it there because it needs to call method from the servicestate package, and that would lead to a cyclic dependency".

Or even removing the comment altogether, as I'm not sure it adds a lot of value.